Xamarin.Forms.Platform.iOS/Material/MaterialFrameRenderer.cs

```diff
@@ -52,6 +52,37 @@ protected override void Dispose(bool disposing)
 			base.Dispose(disposing);
 		}
 
+		public override void WillRemoveSubview(UIView uiview)
+		{
+			var content = Element?.Content;
+			if (content != null && uiview == Platform.GetRenderer(content))
+			{
+				uiview.Layer.Mask = null;
+			}
+
+			base.WillRemoveSubview(uiview);
+		}
+
+		public override void LayoutSubviews()
+		{
+			base.LayoutSubviews();
+
+			var content = Element?.Content;
+			if (content != null && Layer is ShapedShadowLayer shadowLayer)
+			{
+				var renderer = Platform.GetRenderer(content);
+				if (renderer is UIView uiview)
+				{
+					var padding = Element.Padding;
+					var offset = CGAffineTransform.MakeTranslation((nfloat)(-padding.Left), (nfloat)(-padding.Top));
+					uiview.Layer.Mask = new CAShapeLayer
+					{
+						Path = new CGPath(shadowLayer.ShapeLayer.Path, offset)
+					};
+				}
+			}
+		}
```

Comment:
This chunk is for properly clipping the frame contents to the card's rounded corners.